src/commands/runDraftTool/runBuildOnAcr.ts

```diff
@@ -0,0 +1,77 @@
+import {KnownRunStatus} from '@azure/arm-containerregistry';
+import {buildAcrImage} from '../../utils/dockerExtension';
+import {longRunning} from '../../utils/host';
+import {State, StateApi} from '../../utils/state';
+import {Context} from './model/context';
+import {parseAzureResourceId} from '@microsoft/vscode-azext-azureutils';
+import * as vscode from 'vscode';
+import {image} from '../../utils/acr';
+import {CompletedSteps} from './model/guidedExperience';
+
+export async function runBuildAcrImage(
+   {actionContext, extensionContext}: Context,
+   completedSteps: CompletedSteps
+) {
+   const state: StateApi = State.construct(extensionContext);
+
+   // this will avoid prompting user to select which Dockerfile to build if one was just Drafted
+   let chosenDockerfile: vscode.Uri | undefined = undefined;
+   const dockerfile = state.getDockerfile();
+   if (completedSteps.draftDockerfile && dockerfile) {
+      chosenDockerfile = vscode.Uri.file(dockerfile);
+   }
+
+   const run = await longRunning('Building ACR Image', () =>
+      buildAcrImage(chosenDockerfile)
+   );
+   if (run === undefined) {
```

Comment:
For [Undefined Responses](https://github.com/microsoft/vscode-docker/pull/3695#pullrequestreview-1176459780) from the docker extension, it might be beneficial to show a dialog box stating we couldn't process the request successfully

Reply:
It's impossible to distinguish from the case where the user cancels the experience (will return undefined) and the linked case where an error is caught by the docker extension.

In the second case the Docker extension will show a dialog box explaining the failure. I don't think we need to show a second one. If there's a failure in the Docker extension side they will appropriately handle and display the reason.

Reply:
isn't there a way if user press esc button, doesn't the ui component notify this is user cancelled? Just curious. It would be great if we are able to differentiate them.

Reply:
The issue is the escaping is happening in the Docker extension which will return `undefined`. Theoretically we could detect that they are escaping in the Docker extension and make changes to what they'd return but that's unnecessary.

Whenever the Docker extension returns `undefined` we can simply exit our experience like this code does. They will return `undefined` in two cases.
1. The user exits the experience. It makes sense to also exit our experience in this case.
2. There was an error inside the Docker extension. In this case the Docker extension will show relevant dialog boxes explaining that the request wasn't processed correctly. We should also exit our experience in this case. We don't need to add logic to this extension creating a dialog box in this case because the Docker extension handles that (no need for two dialog boxes showing the same thing).

Reply:
Agreed, my concern was with the errors but the docker extension handles them, so I'm fine to exit the experience